api/src/test/java/org/openmrs/api/UserServiceTest.java

```diff
@@ -1361,30 +1367,55 @@ public void getUserByEmail_shouldGetUserByUsingEmail() {
 		User user = userService.getUserByEmail("hank.williams@gmail.com");
 		assertNotNull("User with email hank.williams@gmail not found in database", user);
 	}
-	/*
-	 * @see UserService#getUserByActivationKey(String)
-	 */
+	
 	@Test
-	public void getUserByActivationKey_shouldGetUserUsingActivationKey() {
-		executeDataSet(XML_FILENAME);
-		String key="XbklsieiskeuztecEnPf";
-		User user = userService.getUserByActivationKey(key);	
-		assertNotNull("User with activationKey ", user); 
-		String[] tokens = user.getActivationKey().split(":");
-		assertEquals(key,tokens[0]);		
-
+	public void setUserActivationKey_shouldCreateUserActivationKey() {
+		User u = new User();
+		u.setPerson(new Person());
+		u.addName(new PersonName("Benjamin", "A", "Wolfe"));
+		u.setUsername("bwolfe");
+		u.getPerson().setGender("M");
+		User createdUser = userService.createUser(u, "Openmr5xy");
+		userService.setUserActivationKey(createdUser);
+		String activationKey  = dao.getLoginCredential(createdUser).getActivationKey();
```

Comment:
You should instead be asserting that UserService.getUserActivationKey returns the same user and not this dao call

Reply:
Did you see this comment?